framework/src/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -980,69 +980,40 @@ public boolean equals(Object o) {
     @SideEffectFree
     @Override
     public String toString() {
-        return toDOToutput().replace("\\n", "\n");
+        return "Use a CFGVisualizer to see the Store: " + this.hashCode();
     }
 
-    @Pure
-    @Override
-    public boolean hasDOToutput() {
-        return true;
-    }
-
-    /**
-     * @return DOT representation of the store (may contain control characters
-     *         such as "\n").
-     */
+    @SuppressWarnings("unchecked")
     @Override
-    public String toDOToutput() {
-        StringBuilder result = new StringBuilder(this.getClass()
-                .getCanonicalName() + " (\\n");
-        internalDotOutput(result);
-        result.append(")");
-        return result.toString();
-    }
-
-
-    protected String escapeDoubleQuotes(final String str) {
-        return str.replace("\"", "\\\"");
-    }
-
-
-    protected String toStringEscapeDoubleQuotes(final Object obj) {
-        return escapeDoubleQuotes(String.valueOf(obj));
+	public void visualize(CFGVisualizer<?, S, ?> viz) {
+        ((CFGVisualizer<V, S, ?> ) viz).visualizeStoreHeader(
+            this.getClass().getCanonicalName());
+        internalVisualize((CFGVisualizer<V, S, ?>) viz);
+        ((CFGVisualizer<V, S, ?> ) viz).visualizeStoreFooter();
     }
 
     /**
      * Adds a DOT representation of the internal information of this store to
      * {@code result}.
      */
-    protected void internalDotOutput(StringBuilder result) {
+    protected void internalVisualize(CFGVisualizer<V, S, ?> viz) {
         for (Entry<FlowExpressions.LocalVariable, V> entry : localVariableValues.entrySet()) {
-            result.append("  " + entry.getKey() + " > " + toStringEscapeDoubleQuotes(entry.getValue())
-                    + "\\n");
+            viz.visualizeLocalVariable(entry.getKey(), entry.getValue());
         }
         if (thisValue != null) {
-            result.append("  this > " + thisValue
-                    + "\\n");
+            viz.visualizeThisValue(thisValue);
         }
-        for (Entry<FlowExpressions.FieldAccess, V> entry : fieldValues
-                .entrySet()) {
-            result.append("  " + entry.getKey() + " > " + toStringEscapeDoubleQuotes(entry.getValue())
-                    + "\\n");
+        for (Entry<FlowExpressions.FieldAccess, V> entry : fieldValues.entrySet()) {
+            viz.visualizeFieldValues(entry.getKey(), entry.getValue());
         }
-        for (Entry<FlowExpressions.ArrayAccess, V> entry : arrayValues
-                .entrySet()) {
-            result.append("  " + entry.getKey() + " > " + toStringEscapeDoubleQuotes(entry.getValue())
-                    + "\\n");
+        for (Entry<FlowExpressions.ArrayAccess, V> entry : arrayValues.entrySet()) {
+            viz.visualizeArrayAccess(entry.getKey(), entry.getValue());
```

Comment:
I would rename visualizeArrayAccess to visualizeArrayValue, to be consistent with fields and pure methods.